### DISCUSSION TITLE
remarks on using ultorg

### DISCUSSION THREAD

**Discussion by malcook** [ultorg/public_issues#49] [General]
My initial use of ultorg has tapered off.  I really like it and could see it taking off.  For my current development needs, I am reaching for DB Beaver (which doesn't try and sanitize column/table names).

I'm not going to submit separate issues for these but offer these general remarks.

 - formatting of column headers (e.g. splitting on non-word) very nice except: would like an option to keep ALL CAPS words 
 - `tools > platform options`  would be more easily discoverable as `tools > options` or `edit > options` (after all, the options dialog is merely titled “options”, not “platform options”)
 -  keymap is very handy - thanks
 - show as html - should arguably respect `tools > platform options > general ? web browser` 
 - `help` there is none!  I looked for application documentation/user guid on your website to link to from your help menu but found none there either.
 - synchronization of selected column with Fields is quite nice and could profitably work in both directions (e.g. selecting from Fields could shift selected column left/right as needed)
 - `Unexpected Exception > Show and Report Problem` does not allow to report problem.
 - Missing Features
    - weak support for keyboard alternatives
    - tab key focus navigation missing
    - Filter by selected cell’s value

**Comment by eirikbakke** [comment]
Thank you for these and earlier detailed comments!

I'll try to rethink the feature that auto-formats column names (e.g. "customer_id" becoming "Customer ID"), and see what settings should exist and at what level. (As we also discussed at https://github.com/ultorg/public_issues/issues/26 .) The comment about all caps words is noted; right now there's just a dictionary of common abbreviations that should be capitalized ("ID", "XML", "FK" etc.).

> Filter by selected cell’s value

Yep, good idea. I need to adjust the way selection works; right now the context menu only works for actions that apply to the entire column.

> help there is none! I looked for application documentation/user guid on your website to link to from your help menu but found none there either.

Indeed! Proper documentation pages need to be written. And the main window menus have still not been filled in with the intended actions.

> tab key focus navigation missing

Are you referring to the main window, or in specific dialogs? In the main Windows, the tab key works like in Excel, though Ctrl+Tab can be used to switch between tabs and sidebars. For field selection, it's quickest to hit Command+K (or Ctrl+K on Windows) and then use arrow keys and space to make selections. Same for filter.

**Comment by malcook** [comment]
Thanks for your considered reply.

The next time I reach for ultorg, I will revisit how tab key focus works with your description below in mind.  I think I missed something earlier.

That said, I think the best first doc you could provide would be a complete shortcut keys cheat-sheet (being most helpful to power users most likely to give you good UX feedback).

Cheers,

Malcolm

**Comment by eirikbakke** [comment]
There's another Ultorg release available today (Ultorg 1.2.7), which adds a toolbar button to quickly turn auto-formatting of column names on or off at the perspective level:

![preferTechnicalNames](https://user-images.githubusercontent.com/886243/229630016-c71e7c1f-0c63-44c7-8434-9790b6b039ab.png)

I decided to include this setting at the perspective level at first, since it is useful as a formatting setting. Later I might add separate settings at the data souce level (e.g. to specify that auto-formatting of column names should never be applied for a particular data source, or to customize some aspect of the auto-formatting).

> That said, I think the best first doc you could provide would be a complete shortcut keys cheat-sheet (being most helpful to power users most likely to give you good UX feedback).

Noted and in progress!

(Existing Ultorg users can find the latest version at the old download link, or email me at ebakke@ultorg.com for a new link. The full release notes are at https://www.ultorg.com/releasenotes/ as usual.)
